src/main/java/pl/multiplex/multiplexreservation/api/ReservationApi.java

```diff
@@ -1,4 +1,69 @@
 package pl.multiplex.multiplexreservation.api;
 
-public class ReservationApi {
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import pl.multiplex.multiplexreservation.dao.entity.Movie;
+import pl.multiplex.multiplexreservation.dao.entity.Screening;
+import pl.multiplex.multiplexreservation.dao.entity.Seat;
+import pl.multiplex.multiplexreservation.dao.entity.User;
+import pl.multiplex.multiplexreservation.manager.ReservationManager;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("multiplex/reservation")
+public class ReservationApi
+{
+    private ReservationManager reservations;
+
+
+    @Autowired
+    public ReservationApi(ReservationManager reservation) {
+        this.reservations = reservation;
+    }
+
+    @GetMapping("/screening/all")
+    public Iterable<Screening> getAll() {return reservations.showAll();}
+
+    @GetMapping("/screening")
+    public List<Screening> getByDayAndTime(@RequestParam int date, int time) {
+        return reservations.showByTimeAndDay(date,time);
+    }
+
+    @GetMapping("/screening/interval")
+    public List<Screening> getByDayAndMinMaxTime(@RequestParam int date, int timeMin, int timeMax) {
+        return reservations.showByTimeAndMinMaxDay(date,timeMin, timeMax);
+    }
+
+    @GetMapping("/screening/titles")
+    public List<String> getTitlesByDayAndMinMaxTime(@RequestParam int date, int timeMin, int timeMax) {
+        return reservations.showByTimeAndMinMaxDayOnlyTitle(date,timeMin, timeMax);
+    }
+
+    @GetMapping("/chooseseat")
+    public List<Seat> getSeatsByScreeningId(@RequestParam Long screeningId) {
+        List<Seat> seats = reservations.getSeatsByScreeningId(screeningId);
+        return seats;
```

Comment:
Czemu nie inline tej zmiennej?

Reply:
Szczerze, nigdy nie robiłem niczego za pomocą inline, więc nie przyszło mi to do głowy. Ale chętnie się naucze jak działa!

Reply:
Inline zmiennej - to znaczy w jej miejsce wstawić kod, np zamiast:
```
List<Seat> seats = reservations.getSeatsByScreeningId(screeningId);
return seats;
```
Po prostu:
```
return reservations.getSeatsByScreeningId(screeningId);
```

Reply:
aaaaa, no to rozumiem o co chodzi i często tego używam. Tutaj pewnie coś się w poprzednich wersjach kodu działo pomiędzy i potem nie poprawiłem. Ale ogólnie zdecydowanie by to wyglądało lepiej i byłoby wydajniejsze